platform/security/common/src/main/java/ddf/security/samlp/impl/SamlValidator.java

```diff
@@ -287,6 +288,20 @@ private Request(Builder builder) {
       logoutRequest = (LogoutRequest) builder.xmlObject;
     }
 
+    /**
+     * A SAML LogoutRequest must include at least one SessionIndex, if the logout is initiated by
+     * Session Participant. However, if the Session Authority initiates the logout, a SessionIndex
+     * is optional. In that case, the absence of a SessionIndex means to logout of all sessions
+     * associated with the Principle.
+     *
+     * @throws ValidationException
+     */
+    protected void checkSessionIndexes() throws ValidationException {
+      if (logoutRequest.getSessionIndexes().isEmpty()) {
+        throw new ValidationException("Logout request does not contain a session index");
+      }
+    }
+
```

Comment:
In hindsight, this check might not be the best long-term solution. See the comment at the top of the method. Also, The only kind of requests passed to the SamlValidator are logout requests. If other kinds of requests, like queries or auth, they would incorrectly fail validation.